### PR TITLE
libtxt: have GetRectsForRange(strut) fall back to tight bounds if layout isn't forcing use of the strut

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: third_party/txt/src/txt/paragraph.cc [hunk 1]
```diff
@@ -1458,12 +1458,17 @@ std::vector<Paragraph::TextBox> Paragraph::GetRectsForRange(
                            box.direction);
       }
     } else if (rect_height_style == RectHeightStyle::kStrut) {
-      for (const Paragraph::TextBox& box : kv.second.boxes) {
-        boxes.emplace_back(
-            SkRect::MakeLTRB(
-                box.rect.fLeft, line_baselines_[kv.first] - strut_.ascent,
-                box.rect.fRight, line_baselines_[kv.first] + strut_.descent),
-            box.direction);
+      if (strut_.force_strut) {
```
Comment (GaryQian) [inline third_party/txt/src/txt/paragraph.cc]: Try `paragraph_style_.strut_enabled && paragraph_style_.strut_font_size >= 0` as the condition?

Comment (jason-simmons) [reply third_party/txt/src/txt/paragraph.cc]: Updated this to use the same criteria as `valid_strut` in `ComputeStrut`

File: third_party/txt/src/txt/paragraph.cc [hunk 2]
```diff
@@ -1458,12 +1461,17 @@ std::vector<Paragraph::TextBox> Paragraph::GetRectsForRange(
                            box.direction);
       }
     } else if (rect_height_style == RectHeightStyle::kStrut) {
-      for (const Paragraph::TextBox& box : kv.second.boxes) {
-        boxes.emplace_back(
-            SkRect::MakeLTRB(
-                box.rect.fLeft, line_baselines_[kv.first] - strut_.ascent,
-                box.rect.fRight, line_baselines_[kv.first] + strut_.descent),
-            box.direction);
+      if (IsStrutValid()) {
```
Comment (GaryQian) [inline third_party/txt/src/txt/paragraph.cc]: Can you document this fallback behavior here as well as in the enum declaration?

**Comment by GaryQian** [comment]
Please add a test to verify that disabled strut returns tight bounds

**Comment by jason-simmons** [comment]
Done - PTAL

**Pull request merged**